test_conformance/extensions/cl_khr_command_buffer/negative_command_buffer_copy.cpp

```diff
@@ -0,0 +1,503 @@
+        clMemWrapper image = create_image_2d(
+            context, CL_MEM_READ_WRITE, &formats, 512, 512, 0, NULL, &error);
+        test_error(error, "create_image_2d failed");
+        clMemWrapper buffer = clCreateBuffer(context, CL_MEM_READ_WRITE,
+                                             data_size(), nullptr, &error);
+        test_error(error, "Unable to create buffer");
+
+        error = clCommandCopyImageToBufferKHR(nullptr, nullptr, image, buffer,
+                                              origin, region, 0, 0, nullptr,
+                                              nullptr, nullptr);
+
+        test_failure_error_ret(error, CL_INVALID_OPERATION,
+                               "clCommandCopyImageToBufferKHR should return "
+                               "CL_INVALID_OPERATION",
+                               TEST_FAIL);
+
+        return CL_SUCCESS;
+    }
+};
+
+// CL_INVALID_VALUE if mutable_handle is not NULL.
+struct CommandBufferCopyMutableHandleNotNull : public BasicCommandBufferTest
+{
+    using BasicCommandBufferTest::BasicCommandBufferTest;
+
+    cl_int Run() override
+    {
+        cl_mutable_command_khr mutable_handle;
+
+        cl_int error = clCommandCopyBufferKHR(
+            command_buffer, nullptr, in_mem, out_mem, 0, 0, data_size(), 0,
+            nullptr, nullptr, &mutable_handle);
+
+        test_failure_error_ret(error, CL_INVALID_VALUE,
+                               "clCommandCopyBufferKHR should return "
+                               "CL_INVALID_VALUE",
+                               TEST_FAIL);
+
+        const size_t origin[3] = { 0, 0, 0 }, region[3] = { 512, 512, 1 };
+
+        error = clCommandCopyBufferRectKHR(
+            command_buffer, nullptr, in_mem, out_mem, origin, origin, region, 0,
+            0, 0, 0, 0, nullptr, nullptr, &mutable_handle);
+
+        test_failure_error_ret(error, CL_INVALID_VALUE,
+                               "clCommandCopyBufferRectKHR should return "
+                               "CL_INVALID_VALUE",
+                               TEST_FAIL);
+
+        const cl_image_format formats = { CL_RGBA, CL_UNSIGNED_INT8 };
+        clMemWrapper image = create_image_2d(
+            context, CL_MEM_READ_WRITE, &formats, 512, 512, 0, NULL, &error);
+        test_error(error, "create_image_2d failed");
+        clMemWrapper buffer = clCreateBuffer(context, CL_MEM_READ_WRITE,
+                                             data_size(), nullptr, &error);
+        test_error(error, "Unable to create buffer");
+
+        error = clCommandCopyImageToBufferKHR(
+            command_buffer, nullptr, image, buffer, origin, region, 0, 0,
+            nullptr, nullptr, &mutable_handle);
+
+        test_failure_error_ret(error, CL_INVALID_VALUE,
+                               "clCommandCopyImageToBufferKHR should return "
+                               "CL_INVALID_VALUE",
+                               TEST_FAIL);
+
+        return CL_SUCCESS;
+    }
+};
+};
```

Comment:
trailing semicolon ?

Reply:
@silverclaw Yes, it is closing anonymous namespace, I have done it the same way as other tests are done in this directory.